misk/src/main/kotlin/misk/web/actions/ReadinessCheckAction.kt

```diff
@@ -0,0 +1,42 @@
+package misk.web.actions
+
+import com.google.common.util.concurrent.Service
+import misk.healthchecks.HealthCheck
+import misk.web.Get
+import misk.web.JsonResponseBody
+import misk.web.Response
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class ReadinessCheckAction : WebAction {
+    @Inject lateinit var services: MutableSet<Service>
+    @Inject lateinit var healthChecks: MutableSet<HealthCheck>
+
+    @Get("/_readiness")
+    @JsonResponseBody
+    fun readinessCheck(): Response<String> {
+        var isReady = true
+        for (service in services) {
```

Comment:
fwiw you could just do:

``
val isReady = services.all { it.isRunning } && healthChecks.all { it.status().isHealthy }
``

short-circuiting will work properly